modules/jupyter/main.tf

```diff
@@ -63,17 +63,32 @@ module "iap_auth" {
 module "jupyterhub-workload-identity" {
   source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
   use_existing_gcp_sa = !var.create_service_account
-  name                = var.gcp_service_account
+  name                = var.gcp_and_k8s_service_account
   namespace           = var.namespace
   project_id          = var.project_id
-  roles               = ["roles/cloudsql.client", "roles/artifactregistry.reader", "roles/storage.admin"]
-  depends_on = [ module.iap_auth ]
+  roles               = concat(var.predefined_iam_roles, var.gcp_service_account_iam_roles)
```

Comment:
do we need both sets of roles? i think they overlap. can you verify?

Reply:
The first array is the permission we define and the second is user can define more permission, for example I want to my jupyter hub can access to other cluster. Since jupyter notebook may have some new requirement. The reason why I want to have two arrays is because I don't want to user to touch the default permission.